src/cwhy/cwhy.py

```diff
@@ -115,8 +116,13 @@ def __init__(self, diagnostic):
               self.code_locations[(file_name, line_start, line_end)] = abridged_code
               max_code_locations -= 1
 
+        # If the diagnostic didn't come from a context that we know about and
+        # handle in the above loop, we should make sure it's not too long.
+        if not self.code_locations and line == len(diagnostic_lines) - 1:
+            line = min(line, 50)
+
         self.unabridged_diagnostic = '\n'.join(diagnostic_lines) + '\n'
-        self.abridged_diagnostic = '\n'.join(diagnostic_lines[0:line]) + '\n'
+        self.abridged_diagnostic = '```\n' + '\n'.join(diagnostic_lines[0:line]) + '\n```\n'
```

Comment:
Extreme nit on previous code.

```suggestion
        self.abridged_diagnostic = '```\n' + '\n'.join(diagnostic_lines[:line]) + '\n```\n'
```